crates/sui-framework/src/natives/crypto/hash.rs

```diff
@@ -1,50 +1,136 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
-use crate::legacy_empty_cost;
+use crate::natives::NativesCostTable;
 use fastcrypto::hash::{Blake2b256, HashFunction, Keccak256};
 use move_binary_format::errors::PartialVMResult;
-use move_vm_runtime::native_functions::NativeContext;
+use move_core_types::gas_algebra::InternalGas;
+use move_vm_runtime::{native_charge_gas_early_exit, native_functions::NativeContext};
 use move_vm_types::{
     loaded_data::runtime_types::Type,
     natives::function::NativeResult,
     pop_arg,
     values::{Value, VectorRef},
 };
 use smallvec::smallvec;
-use std::collections::VecDeque;
+use std::{collections::VecDeque, ops::Mul};
+
+const BLAKE_2B256_BLOCK_SIZE: u16 = 128;
+const KECCAK_256_BLOCK_SIZE: u16 = 32;
```

Comment:
Are we sure Keccak block size is just 32byte or 136? cc: @benr-ml

Reply:
No, my fault, you are right - it should be 136

Reply:
Got it. Fixed it.